ra2ce/analyses/analysis_config_data/analysis_config_data.py

```diff
@@ -22,24 +22,137 @@
+    uniform_duration: float = math.nan
+    gdp_percapita: float = math.nan
+    equity_weight: str = ""
+    calculate_route_without_disruption: bool = False
+    buffer_meters: float = math.nan
+    threshold_locations: float = math.nan
+    category_field_name: str = ""
+    save_traffic: bool = False
 
+
+@dataclass
+class AnalysisSectionDirect(AnalysisSectionBase):
+    # adaptation/effectiveness measures
+    return_period: float = math.nan
+    repair_costs: float = math.nan
+    evaluation_period: float = math.nan
+    interest_rate: float = math.nan
+    climate_factor: float = math.nan
+    climate_period: float = math.nan
+    # road damage
+    damage_curve: str = ""
+    event_type: str = ""
+    risk_calculation: str = ""
+    create_table: bool = False
+    file_name: Optional[Path] = None
+
+
+@dataclass
 class AnalysisConfigData(ConfigDataProtocol):
-    @classmethod
-    def from_dict(cls, dict_values: dict) -> AnalysisConfigData:
-        _new_analysis_ini_config_data = cls()
-        _new_analysis_ini_config_data.update(**dict_values)
-        return _new_analysis_ini_config_data
+    root_path: Optional[Path] = None
+    input_path: Optional[Path] = None
+    output_path: Optional[Path] = None
+    static_path: Optional[Path] = None
+    project: ProjectSection = field(default_factory=lambda: ProjectSection())
+    analyses: list[AnalysisSectionBase] = field(default_factory=list)
+    files: Optional[dict[str, Path]] = field(default_factory=dict)
+    origins_destinations: Optional[OriginsDestinationsSection] = field(
+        default_factory=lambda: OriginsDestinationsSection()
+    )
+    network: Optional[NetworkConfigData] = field(
+        default_factory=lambda: NetworkConfigData()
+    )
+    hazard_names: Optional[list[str]] = field(default_factory=list)
+
+    def to_dict(self) -> dict:
+        _dict = self.__dict__
+        _dict["project"] = self.project.__dict__
+        _dict["direct"] = [dv.__dict__ for dv in self.direct]
+        _dict["indirect"] = [dv.__dict__ for dv in self.indirect]
+        _dict["files"] = [dv.__dict__ for dv in self.files]
+        return _dict
+
+    @property
+    def direct(self):
+        return list(
+            analysis
+            for analysis in self.analyses
+            if analysis.analysis in DirectAnalysisNameList
+        )
+
+    @property
+    def indirect(self):
+        return list(
+            analysis
+            for analysis in self.analyses
+            if analysis.analysis in IndirectAnalysisNameList
+        )
 
 
 class AnalysisConfigDataWithNetwork(AnalysisConfigData):
-    @classmethod
-    def from_dict(cls, dict_values: dict) -> AnalysisConfigDataWithNetwork:
-        return super().from_dict(dict_values)
+    pass
 
 
 class AnalysisConfigDataWithoutNetwork(AnalysisConfigData):
-    @classmethod
-    def from_dict(cls, dict_values: dict) -> AnalysisConfigDataWithoutNetwork:
-        return super().from_dict(dict_values)
+    pass
```

Comment:
This can (virtually) be removed, possible follow-up subtask?

Reply:
Agree, will put it on the backlog